superset-frontend/plugins/legacy-plugin-chart-map-box/package.json

```diff
@@ -30,7 +30,6 @@
   "dependencies": {
     "@superset-ui/chart-controls": "0.18.25",
     "@superset-ui/core": "0.18.25",
-    "mapbox-gl": "^0.53.0",
```

Comment:
from root `node_module` get this module

Reply:
Could we move these to `peerDependencies` to make sure users of the plugin pull in the relevant deps?

Reply:
`mapbox-gl` is a dependency from the `react-map-gl`, and `mapbox-gl` is a dependency for `deckgl`. The root cause for this issue is that `mapbox-gl` version is incompatible in `legacy-plugin-chart-map-box` and `legacy-preset-chart-deckgl`. 

When we removed `mapbox-gl` in both packages, npm will auto-resolve the conflict. If the user installs those packages as standalone, `mapbox-gl` will install by default.

```
(superset) yongjie.zhao@:superset-frontend$ npm ls --depth 1 | grep -A 30 'map-box'
....
....
├─┬ @superset-ui/legacy-plugin-chart-map-box@0.18.25 -> ./plugins/legacy-plugin-chart-map-box
│ ├── @superset-ui/chart-controls@0.18.25 deduped -> ./packages/superset-ui-chart-controls
│ ├── @superset-ui/core@0.18.25 deduped -> ./packages/superset-ui-core
│ ├── prop-types@15.7.2 deduped
│ ├─┬ react-map-gl@4.1.16
│ │ ├── @babel/runtime@7.15.4 deduped
│ │ ├── mapbox-gl@0.54.1
│ │ ├── mjolnir.js@2.5.0
│ │ ├── prop-types@15.7.2 deduped
│ │ ├── react-virtualized-auto-sizer@1.0.2 deduped
│ │ ├── react@16.14.0 deduped invalid: "^15.5.4" from node_modules/omnibar
│ │ └── viewport-mercator-project@6.2.3 deduped
│ ├── react@16.14.0 deduped
│ ├─┬ supercluster@4.1.1
│ │ └── kdbush@2.0.1
│ └─┬ viewport-mercator-project@6.2.3
│   ├── @babel/runtime@7.15.4 deduped
│   └── gl-matrix@3.3.0
....

```